src/Features/Core/Portable/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.cs

```diff
@@ -87,11 +87,47 @@ protected static string GetOrMapDiagnosticId(Diagnostic diagnostic, out bool inc
             return diagnostic.Id;
         }
 
+        protected abstract SyntaxNode GetContainingStatement(SyntaxToken token);
+        protected abstract bool TokenHasTrailingLineContinuationChar(SyntaxToken token);
+
         protected virtual SyntaxToken GetAdjustedTokenForPragmaDisable(SyntaxToken token, SyntaxNode root, TextLineCollection lines, int indexOfLine)
-            => token;
+        {
+            var containingStatement = GetContainingStatement(token);
+            if (containingStatement is not null && containingStatement.GetFirstToken() != token)
+            {
+                indexOfLine = lines.IndexOf(containingStatement.GetFirstToken().SpanStart);
+                var line = lines[indexOfLine];
+                token = root.FindToken(line.Start);
+            }
+
+            return token;
+        }
+
+        private SyntaxToken GetAdjustedTokenForPragmaRestore(SyntaxToken token, SyntaxNode root, TextLineCollection lines, int indexOfLine)
+        {
+            var containingStatement = GetContainingStatement(token);
```

Comment:
oh definitely doc what's going on here.  def not sure what it means to get the containing statement, but then have this loop that is walking tokens and comparing them to it.